src/xcode/ENA/ENA/Source/Services/DownloadedPackagesStore/V1/DownloadedPackagesSQLLiteStoreV1.swift

```diff
@@ -174,44 +173,39 @@ extension DownloadedPackagesSQLLiteStoreV1: DownloadedPackagesStoreV1 {
 
 			guard deleteHours() else {
 				self.database.rollback()
-				self.failAsyncWithError(completion: completion, errorCode: database.lastErrorCode())
 				return
 			}
 			guard insertDay() else {
 				self.database.rollback()
-				self.failAsyncWithError(completion: completion, errorCode: database.lastErrorCode())
 				return
 			}
 
 			self._commit()
-
-			self.completeAsync(completion: completion)
 		}
 
-	}
-
-	private func failAsyncWithError(completion: ((SQLiteErrorCode?) -> Void)?, errorCode: Int32) {
-		DispatchQueue.global().async {
-			if let error = SQLiteErrorCode(rawValue: errorCode) {
-				completion?(error)
-			} else {
-				completion?(.unknown)
-			}
+		let lastErrorCode = database.lastErrorCode()
+		if lastErrorCode == 0 {
+			return .success(())
+		} else {
+			return .failure(error(for: lastErrorCode))
 		}
 	}
 
-	private func completeAsync(completion: ((SQLiteErrorCode?) -> Void)?) {
-		DispatchQueue.global().async {
-			completion?(nil)
+	private func error(for sqliteErrorCode: Int32) -> SQLiteErrorCode {
+		if let error = SQLiteErrorCode(rawValue: sqliteErrorCode) {
+			return error
+		} else {
+			return .unknown
 		}
 	}
 
+	@discardableResult
 	func set(
 		country: Country.ID,
 		hour: Int,
 		day: String,
 		package: SAPDownloadedPackage
-	) {
+	) -> Result<Void, SQLiteErrorCode> {
```

Comment:
Whats the benefit?

Reply:
Cleaner code.

I'd prefer `try set(…)` over `_ = store.set(country: country, day: date, package: package)` or simply `set(…)` because the `try` enforces some proper error handling and not omitting most of it.

Reply:
The most of the current code is just fire and forget (ignore the outcome).
Adding "throws" could end in a bigger refactoring.
I would like to ask you to create a refactoring story.

Reply:
Will do.